subprojects/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyNotationParser.java

```diff
@@ -19,51 +19,144 @@
+        NotationConverter<MinimalExternalModuleDependency, ? extends MinimalExternalModuleDependency> minimalExternalDependencyNotationConverter =
+            new MinimalExternalDependencyNotationConverter(instantiator, attributesFactory, capabilityNotationParser);
+        MapNotationConverter<? extends ExternalModuleDependency> mapNotationConverter =
+            new DependencyMapNotationConverter<>(instantiator, DefaultExternalModuleDependency.class);
+        NotationConverter<FileCollection, ? extends FileCollectionDependency> filesNotationConverter =
+            new DependencyFilesNotationConverter(instantiator);
+        NotationConverter<Project, ? extends ProjectDependency> projectNotationConverter =
+            new DependencyProjectNotationConverter(dependencyFactory);
+        DependencyClassPathNotationConverter dependencyClassPathNotationConverter = new DependencyClassPathNotationConverter(instantiator, classPathRegistry, fileCollectionFactory, runtimeShadedJarFactory, currentGradleInstallation);
+        NotationParser<Object, Dependency> notationParser = NotationParserBuilder
             .toType(Dependency.class)
             .noImplicitConverters()
-            .fromCharSequence(new DependencyStringNotationConverter<>(instantiator, DefaultExternalModuleDependency.class, stringInterner))
-            .fromType(MinimalExternalModuleDependency.class, new MinimalExternalDependencyNotationConverter(instantiator, attributesFactory, capabilityNotationParser))
-            .converter(new DependencyMapNotationConverter<>(instantiator, DefaultExternalModuleDependency.class))
-            .fromType(FileCollection.class, new DependencyFilesNotationConverter(instantiator))
-            .fromType(Project.class, new DependencyProjectNotationConverter(dependencyFactory))
-            .fromType(DependencyFactory.ClassPathNotation.class, new DependencyClassPathNotationConverter(instantiator, classPathRegistry, fileCollectionFactory, runtimeShadedJarFactory, currentGradleInstallation))
+            .fromCharSequence(stringNotationConverter)
+            .fromType(MinimalExternalModuleDependency.class, minimalExternalDependencyNotationConverter)
+            .converter(mapNotationConverter)
+            .fromType(FileCollection.class, filesNotationConverter)
+            .fromType(Project.class, projectNotationConverter)
+            .fromType(DependencyFactory.ClassPathNotation.class, new NotationConverter<DependencyFactory.ClassPathNotation, Dependency>() {
+                @Override
+                public void convert(DependencyFactory.ClassPathNotation notation, NotationConvertResult<? super Dependency> result) throws TypeConversionException {
+                    DeprecationLogger.deprecateInternalApi("DependencyFactory.ClassPathNotation")
+                        .replaceWith("an appropriate call to DependencyHandler")
+                        .willBeRemovedInGradle8()
+                        .undocumented()
```

Comment:
We shouldn't introduce new undocumented nags.  All we need for this is an entry in the upgrading guide that we can link to.  It can say "Internal API DependencyFactory.ClassPathNotation will be removed. Use <appropriate public APIs>".

Reply:
Done.